### PR TITLE
add integration tests

### PR DESCRIPTION
**Pull request by arufanov**

### CODE REVIEW

**Comment by arufanov** [comment]
@yurichervoniy lets discuss:
* mistakes
* integration test organizaton (test -> unit + system/integration)
* layout ogranization (controller, mapper and etc)
*

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
 class TabrApplicationTests {
 
-  @Test
-  void contextLoads() {
+  @Autowired
+  private MockMvc mockMvc;
 
+  @Test
+  void getHomePage() throws Exception {
+    mockMvc.perform(get("/")).andExpect(status().isOk());
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/TabrApplicationTests.java:23]: What do we test here?

**Comment by yurichervoniy** [comment]
Any suggestions for discuss?
mistakes
integration test organizaton (test -> unit + system/integration)
layout ogranization (controller, mapper and etc)

**Comment by arufanov** [comment]
@yurichervoniy lets discuss:
* mistakes
* integration test organizaton (test -> unit + system/integration)
* layout ogranization (controller, mapper and etc)
*

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
 
+@ExtendWith(SpringExtension.class)
 @SpringBootTest
 @ActiveProfiles("test")
+@AutoConfigureMockMvc
 class TabrApplicationTests {
 
-  @Test
-  void contextLoads() {
+  // @Autowired
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/TabrApplicationTests.java:18]: Remove comments. 
mockMvc - does not starts web server - i think. 
For real IntegrationTest - i think we should consider using TestRestTemplate or TestWebClient

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
 
+@ExtendWith(SpringExtension.class)
 @SpringBootTest
 @ActiveProfiles("test")
+@AutoConfigureMockMvc
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/TabrApplicationTests.java:15]: Pls remove this annotation @AutoConfigureMockMvc from @SpringBootTest

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
 
+@ExtendWith(SpringExtension.class)
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/TabrApplicationTests.java:12]: it's already included in @SpringBootTest

**Review by yurichervoniy — APPROVED** [review]

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java (hunk 1 above)
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/TabrApplicationTests.java]: @yurichervoniy 
* rename file to TabrApplicationIntegrationTests. Test is get root page and get success.  Indeed here is problem to discuss: what should we use: mockMvc has restriction, for example: https://www.baeldung.com/spring-boot-testing has limitation for example redirect. Other approach that I have been already tests is restTemplate, but we have more that rest teamples. see https://stackoverflow.com/questions/25901985/difference-between-mockmvc-and-resttemplate-in-integration-tests. Create independent issue

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java (hunk 2 above)
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/TabrApplicationTests.java]: @yurichervoniy 
see https://github.com/a5lab/tabr/issues/134

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java (hunk 3 above)
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/TabrApplicationTests.java]: Done

File: src/test/java/com/a5lab/tabr/TabrApplicationTests.java (hunk 4 above)
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/TabrApplicationTests.java]: done

File: src/test/java/com/a5lab/tabr/TabrApplicationIntegrationTests.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@ActiveProfiles("test")
+@AutoConfigureMockMvc
+class TabrApplicationIntegrationTests {
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/TabrApplicationIntegrationTests.java:16]: I agree with HomeControllerTests 
But not with TabrApplicationIntegrationTests 
Why do we need this class?

**Review by yurichervoniy — APPROVED** [review]

**Pull request merged**